sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -448,14 +517,19 @@ static int sof_audio_probe(struct platform_device *pdev)
 		sof_rt5682_quirk);
 
 	dai_links = sof_card_dai_links_create(&pdev->dev, sof_rt5682_quirk &
-					      SOF_RT5682_SSP_MASK,
+					      SOF_RT5682_SSP_CODEC_MASK,
+					      sof_rt5682_quirk &
+					      SOF_RT5682_SSP_AMP_MASK,
```

Comment:
I am not sure if it works. SOF_RT5682_SSP_AMP_MASK is GENMASK(8, 6), So  (sof_rt5682_quirk & SOF_RT5682_SSP_AMP_MASK) will be a number in bit 8 to 6, right? Should we modify it to (sof_rt5682_quirk & SOF_RT5682_SSP_AMP_MASK) >> SOF_RT5682_SSP_AMP_SFT? And #define  SOF_RT5682_SSP_AMP_SFT 6